operators/src/pro/cache/cache_tiles.rs

```diff
@@ -296,14 +270,35 @@ impl<T> CacheTileStream<T> {
     }
 }
 
-impl<T: Pixel> Stream for CacheTileStream<T> {
-    type Item = Result<RasterTile2D<T>>;
+impl<T: Pixel> Stream for CacheTileStream<T>
+where
+    RasterTile2D<T>: CacheElement<StoredCacheElement = CompressedRasterTile2D<T>>,
+{
+    type Item = Result<RasterTile2D<T>, CacheError>;
 
     fn poll_next(
         mut self: Pin<&mut Self>,
-        _cx: &mut std::task::Context<'_>,
+        cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<Option<Self::Item>> {
-        let CacheTileStreamProjection { data, query, idx } = self.as_mut().project();
+        let CacheTileStreamProjection {
+            data,
+            query,
+            idx,
+            mut state,
+        } = self.as_mut().project();
+
+        // if there is a tile that is currently being decoded, try to return it
+        if let Some(state_future) = state.as_mut().as_pin_mut() {
+            // futures ready will return pending if the future is not ready yet
+            let tile_res_res = futures::ready!(state_future.poll(cx));
+            state.set(None);
+            let Ok(tile_res) = tile_res_res else {
+                return std::task::Poll::Ready(Some(
+                    Err(CacheError::CouldNotDecompressElement),
```

Comment:
how can this occur? On line 653 you say "Decompression can not fail"

Reply:
now it can since we no prepend the len.